content/posts/2019/2019-07-05-littlest-cuda-hub.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Setting up GPU Data Science environments for hackathons"
+subtitle: "AKA Littlest Cuda Hub"
+date: 2019-07-05T00:00:00+00:00
+draft: true
+author: "Jacob Tomlinson"
+categories:
+  - blog
+tags:
+  - rapids
+  - iris
+  - aws
+  - jupyterhub
+  - cuda
+  - gpu
+thumbnail: jupyter
+---
+
+## Background
+
+In my first week working at Nvidia I have been spending some time with my previous colleagues at the Met Office to explore how the two organisations can collaborate. 
+
+The theme for the week was to run a hackathon to explore how GPUs could be used to accelerate existing tools and workflows within the Met Office. To begin trying things out we needed a collaborative environment with access to GPUs. For this the Met Office kindly provided a GPU EC2 instance on AWS and we set up RAPIDS and the [Littlest JupyterHub](https://tljh.jupyter.org/en/latest/) (tljh) to expose it to collaborators.
+
+## Setup
+
+Let's walk through the steps to create our GPU enabled Littlest JupyterHub.
+
+### Infrastructure
+
+First we will need to build a ubuntu server with GPUs available. As the Met Office folks have provided access to an AWS account we will go through the excellent instructions from the [tljh docs](https://tljh.jupyter.org/en/latest/install/amazon.html) to set up on AWS with a few adjustments below.
+
+- At step 6 we selected a `p3.8xlarge` instance as it comes with four Nvidia Tesla V100 GPUs. 
+- At step 7 we skipped entering the user data (bootstrapping script) as we wanted to run this ourselves manually later.
+- At step 8 we selected 100GB to give ourselves reasonable space for data.
+- Stop after step 16. As we skipped step 7, steps 17-19 do not apply and we need to do some manual configuration before continuing.
+
+### Drivers
+
+Next you need to ssh to the box with the ssh key you selected during setup. In order to make use of GPU enabled libraries we needed to install the NVidia CUDA drivers.
+
+- Visit https://www.nvidia.co.uk/Download/index.aspx
+  - Select the appropriate GPU for the instance (in this case Tesla V100)
+  - Select Linux 64-bit Ubuntu 18.04 
+    - _WARNING: Do not select the generic linux 64-bit driver as it will not work on ubuntu_
+  - Copy the download link and run `wget` on the EC2 instance to download the `.deb` file.
+  - Install your `.deb` file on the server (in our case we ran `dpkg -i nvidia-diag-driver-local-repo-ubuntu1804-418.67_1.0-1_amd64.deb` but your driver file name may differ)
+- This will have installed a new repository but not the actual drivers so let's also do that.
+  - `apt-get update`
+  - `apt-get install cuda-drivers`
+  - We now need to `reboot` for the drivers to be loaded into the kernel
+
+### JupyterHub
+
+Now that we have drivers in place and have rebooted we can install the Littlest JupyterHub as instructed in the documentation. In step 7 we skipped entering the installation command in the user data field, so let's run that manually now.
```

Comment:
An image of a JuptyerHub login screen might be friendly here, even if it's not from our specific deployment.